### PR TITLE
ci: configure codecov

### PR DESCRIPTION
**Pull request by nolouch**
Signed-off-by: nolouch <nolouch@gmail.com>

### What problem does this PR solve?

try to make  `codecov/project` and `codecov/patch` reports more stable. according https://docs.codecov.com/docs/commit-status#branches.

### Check List

Tests

- Unit test

### Release note

```release-note
None
```

### CODE REVIEW

**Review by lhy1024 — APPROVED** [review]

File: codecov.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+coverage:
+  status:
+    project:
+      default:
+        # basic
+        target: auto
+        threshold: 3%
+        base: auto
```
Comment (JmPotato) [inline codecov.yml:8]:
```
base
[Deprecated as of July 2020 -- Codecov's product roadmap includes a novel way to manually pick base of pull requests]
```

According to https://docs.codecov.com/docs/commit-status#base, this parameter seems unnecessary.

**Review by JmPotato — COMMENTED** [review]
Rest LGTM.

**Review by JmPotato — APPROVED** [review]

File: codecov.yml [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+    project:
+      default:
+        # basic
+        target: auto
+        threshold: 3%
+    patch:
+      default:
+        # basic
+        target: auto
+        threshold: 3%
```
Comment (rleungx) [inline codecov.yml:12]: leave an empty line?

**Pull request merged**